virttest/test_setup.py

```diff
@@ -1301,6 +1300,26 @@ def get_pf_ids(self):
             pf_ids.append(pf_id)
         return pf_ids
 
+    def assign_static_ip(self):
+        """
+        Set the static IP for the PF devices for mlx5_core driver
+        """
+        # This function assigns static IP for the PF devices
+        pf_devices = self.get_pf_ids()
+        if (not self.start_addr_PF) or (not self.net_mask):
+            raise exceptions.TestSetupFail(
+                "No IP / netmask found, please populate starting IP address for PF devices in configuration file")
+        ip_addr = netaddr.IPAddress(self.start_addr_PF)
+        for PF in pf_devices:
+            ifname = utils_misc.get_interface_from_pci_id(PF)
+            ip_assign = "ifconfig %s %s netmask %s up" % (
+                ifname, ip_addr, self.net_mask)
+            logging.info("assign IP to PF device %s : %s", PF,
+                         ip_assign)
+            process.system(ip_assign, shell=True, ignore_status=True)
```

Comment:
please have a check on the return value and raise `TestSetupFail()` if command fails

Reply:
@balamuruhans Incorporated the changes suggested, please review. Thanks.